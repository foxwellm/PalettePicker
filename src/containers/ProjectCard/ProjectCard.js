import React from 'react';

export const ProjectCard = (props) => {
  
    const palettes = props.palettes.map(palette => {
      return <div className='card-palette'>
        <span>{palette.name}</span>
        <div style={{ backgroundColor: palette.color1 }} className='palette-color-div'></div>
        <div style={{ backgroundColor: palette.color2 }} className='palette-color-div'></div>
        <div style={{ backgroundColor: palette.color3 }} className='palette-color-div'></div>
        <div style={{ backgroundColor: palette.color4 }} className='palette-color-div'></div>
        <div style={{ backgroundColor: palette.color5 }} className='palette-color-div'></div>
      </div>
    });

    return (
      <div className='project-card'>
        <h1>{props.projectTitle}</h1>
        {palettes}
      </div>
    )
  
}