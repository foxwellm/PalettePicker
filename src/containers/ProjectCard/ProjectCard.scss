.project-card {
  width: 400px;
  border: 1px solid black;
  border-radius: 10px;
  padding: 10px 20px;
  box-shadow: 1px 1px 5px grey;
}

.card-palette {
  display: flex;
}

.palette-color-div {
  width: 30px;
  height: 30px;
  margin: 3px;
}

.delete-project {
  position: relative;
  top: -40px;
  right: -340px;
  height: 20px;
  width: 20px;
  background: url('./images/trash.png');
  background-size: cover;
  border: none;
  cursor: pointer;

  &:hover {
    background: url('./images/trash-hover.png');
    background-size: cover;
  }
}