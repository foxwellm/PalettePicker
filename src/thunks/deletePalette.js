import { setLoading, setError, removePalette } from '../actions';
import { fetchData } from '../utils/api';

export const deletePalette = (id) => {
  return async (dispatch) => {
    try {
      dispatch(setLoading(true));
      await fetchData(`/api/v1/palettes/${id}`, 'DELETE');
      dispatch(removePalette(id));
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false));
  }
}