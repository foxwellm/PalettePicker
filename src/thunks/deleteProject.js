import { setLoading, setError, removeProject } from '../actions';
import { fetchData } from '../utils/api';

export const deleteProject = (id) => {
  return async (dispatch) => {
    try {
      dispatch(setLoading(true));
      await fetchData(`/api/v1/projects/${id}`, 'DELETE');
      dispatch(removeProject(id));
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false));
  }
}