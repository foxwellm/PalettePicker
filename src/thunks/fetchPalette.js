import { setLoading, setError, setPalettes } from '../actions';
import { fetchData } from '../utils/api';

export const fetchPalette = (id) => {
  return async (dispatch) => {
    try {
      dispatch(setLoading(true));
      const palette = await fetchData(`/api/v1/palettes/${id}`, 'GET');
      dispatch(setPalettes([palette]));
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false));
  };
};