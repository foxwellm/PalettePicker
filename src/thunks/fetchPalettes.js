import { setError } from '../actions';
import { fetchData } from '../utils/api';

export const fetchPalettes = (allProjects) => {
  return (dispatch) => {
    const promisedPalettes = allProjects.map(async palette => {
      try {
        let allPalettes = await fetchData(`/api/v1/projects/${palette.id}/palettes`, 'GET');
        allPalettes.sort((a, b) => {
          if (a.updated_at > b.updated_at) return -1;
          if (b.updated_at < a.updated_at) return 1;
          return 0;
        });
        return allPalettes
      } catch (error) {
        dispatch(setError(error.message));
      }
    })
    return Promise.all(promisedPalettes);
  }
}