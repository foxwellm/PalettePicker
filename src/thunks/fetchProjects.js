import { setLoading, setError, setProjects } from '../actions';
