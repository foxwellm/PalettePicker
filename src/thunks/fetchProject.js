import { setLoading, setError, setProjects, setPalettes } from '../actions';
import { fetchData } from '../utils/api';
import { fetchPalettes } from './fetchPalettes';

export const fetchProject = (id) => {
  return async (dispatch) => {
    try {
      dispatch(setLoading(true));
      const project = await fetchData(`/api/v1/projects/${id}`, 'GET');
      dispatch(setProjects([project]));
      const projectPalettes = await dispatch(fetchPalettes([project]));
      dispatch(setPalettes(projectPalettes));
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false));
  };
};