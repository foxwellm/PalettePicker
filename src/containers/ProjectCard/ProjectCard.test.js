import React from 'react';
import { ProjectCard, mapDispatchToProps } from './ProjectCard';
import { shallow } from 'enzyme';
import { deleteProject } from '../../thunks/deleteProject';
jest.mock('../../thunks/deleteProject');

describe('ProjectCard', () => {
  let wrapper;
  let mockDeleteProject;
  let mockPalettes;
  let mockProjectTitle;
  let mockId;
  let mockEvent;

  beforeEach(() => {
    mockEvent = { target: { id: '1' } };
    mockDeleteProject = jest.fn();
    mockPalettes = [{
      id: 1,
      name: 'Palette 1',
      color1: '#ffffff',
      color2: '#ffffff',
      color3: '#ffffff',
      color4: '#ffffff',
      color5: '#ffffff'
    }];

    wrapper = shallow(
      <ProjectCard
        deleteProject={mockDeleteProject}
        id={mockId}
        projectTitle={mockProjectTitle}
        palettes={mockPalettes}
      />
    )
  });

  it('should match the correct snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it.skip('should call deleteProject on click when deleteCard is called', async () => {
    await wrapper.find('button').simulate('click');
    expect(mockDeleteProject).toHaveBeenCalledWith(parseInt(mockEvent.target.id));
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch when deleteProject is called', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = deleteProject();
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.deleteProject();
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });

});