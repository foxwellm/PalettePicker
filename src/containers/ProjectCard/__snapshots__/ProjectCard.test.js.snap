// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectCard should match the correct snapshot 1`] = `
<div
  className="project-card"
>
  <h3 />
  <button
    className="delete-project"
    onClick={[Function]}
  />
  <Connect(ProjectCardPalette)
    color1="#ffffff"
    color2="#ffffff"
    color3="#ffffff"
    color4="#ffffff"
    color5="#ffffff"
    id={1}
    key="1"
    name="Palette 1"
  />
</div>
`;
